src/multiagent-system/modules/melodyPlayer/melodyPlayer.worker.ts

```diff
@@ -1,23 +1,28 @@
 import WebWorker from '@mas/modules/common/models/webWorker';
 import { Point } from '@mas/modules/common/models/point';
-import Note from './models/note';
-import PianoKey from './models/pianoKey';
+import { Note, INote } from './models/note';
+import { PianoKey ,IPianoKey } from './models/pianoKey';
 import { MelodyBall } from './models/units/melodyBall';
 
 export default {} as typeof Worker & (new () => Worker);
 
+interface MelodyPlayerWorkerData {
+    melody: Array<INote>;
+    pianoKeys: Array<IPianoKey>;
+}
+
 class MelodyPlayerWorker extends WebWorker<MelodyBall> {
     melody: Array<Note>;
-    currentTime: number;
     pianoKeys: Array<PianoKey>;
+    currentTime: number;
 
     constructor() {
         super();
     }
 
-    setInitialData(initialData: any): void {
-        this.melody = initialData.melody;
-        this.pianoKeys = initialData.pianoKeys;
+    setInitialData(initialData: MelodyPlayerWorkerData): void {
+        this.melody = initialData.melody.map(note => new Note(note));
+        this.pianoKeys = initialData.pianoKeys.map(key => new PianoKey(key));
```

Comment:
Will prevent errors in future if `Note` and `PianoKey` will get any methods. If they will not, consider to use just and interface instead of a class.

Reply:
U are awesome